### ISSUE TITLE
Exceptions in bare control code trigger stacktrace in inspec 1.x

### ISSUE THREAD

**Issue by jeremymv2** [inspec/inspec#3094]
### 🎛 Description

With the following profile:
```
control "my_control" do
   only_if { mount('/tmp').mounted? }
   describe mount('/tmp') do
     it { should be_mounted }
   end
end
```
Running `inspec archive /path/to/profile` with latest 1.x version (1.51.25) results in this stacktrace:

```
/Library/Ruby/Gems/2.3.0/gems/inspec-1.51.25/lib/resources/mount.rb:32:in `mounted?': undefined method `mounted?' for nil:NilClass (NoMethodError)
	from test_profile/controls/example.rb:2:in `block (2 levels) in load_with_context'
	from /Library/Ruby/Gems/2.3.0/gems/inspec-1.51.25/lib/inspec/rule.rb:110:in `only_if'
	from test_profile/controls/example.rb:2:in `block in load_with_context'
	from /Library/Ruby/Gems/2.3.0/gems/inspec-1.51.25/lib/inspec/rule.rb:50:in `instance_eval'
	from /Library/Ruby/Gems/2.3.0/gems/inspec-1.51.25/lib/inspec/rule.rb:50:in `initialize'
	from /Library/Ruby/Gems/2.3.0/gems/inspec-1.51.25/lib/inspec/control_eval_context.rb:75:in `new'
	from /Library/Ruby/Gems/2.3.0/gems/inspec-1.51.25/lib/inspec/control_eval_context.rb:75:in `block (2 levels) in create'
	from test_profile/controls/example.rb:1:in `load_with_context'
	from /Library/Ruby/Gems/2.3.0/gems/inspec-1.51.25/lib/inspec/profile_context.rb:157:in `instance_eval'
	from /Library/Ruby/Gems/2.3.0/gems/inspec-1.51.25/lib/inspec/profile_context.rb:157:in `load_with_context'
	from /Library/Ruby/Gems/2.3.0/gems/inspec-1.51.25/lib/inspec/profile_context.rb:141:in `load_control_file'
	from /Library/Ruby/Gems/2.3.0/gems/inspec-1.51.25/lib/inspec/profile.rb:167:in `block in collect_tests'
	from /Library/Ruby/Gems/2.3.0/gems/inspec-1.51.25/lib/inspec/profile.rb:164:in `each'
	from /Library/Ruby/Gems/2.3.0/gems/inspec-1.51.25/lib/inspec/profile.rb:164:in `collect_tests'
	from /Library/Ruby/Gems/2.3.0/gems/inspec-1.51.25/lib/inspec/profile.rb:470:in `load_checks_params'
	from /Library/Ruby/Gems/2.3.0/gems/inspec-1.51.25/lib/inspec/profile.rb:463:in `load_params'
	from /Library/Ruby/Gems/2.3.0/gems/inspec-1.51.25/lib/inspec/profile.rb:157:in `params'
	from /Library/Ruby/Gems/2.3.0/gems/inspec-1.51.25/lib/inspec/profile.rb:323:in `controls_count'
	from /Library/Ruby/Gems/2.3.0/gems/inspec-1.51.25/lib/inspec/profile.rb:294:in `check'
	from /Library/Ruby/Gems/2.3.0/gems/inspec-1.51.25/lib/inspec/cli.rb:145:in `archive'
	from /Library/Ruby/Gems/2.3.0/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
	from /Library/Ruby/Gems/2.3.0/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
	from /Library/Ruby/Gems/2.3.0/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
	from /Library/Ruby/Gems/2.3.0/gems/thor-0.20.0/lib/thor/base.rb:466:in `start'
	from /Library/Ruby/Gems/2.3.0/gems/inspec-1.51.25/bin/inspec:12:in `<main>'
```

### 👑 Reason for concern
This affects Chef Automate v1 customers since we have InSpec appbundled with version locked to `1.51.18` meaning they cannot upload compliance profiles utilizing the `mount.mounted?` method.

### 🌍 InSpec and Platform Version

When using InSpec 1.x on Mac, Centos, Ubuntu, RHEL:

* For the above example profile, inspec `check`, `exec` works, however `archive` does NOT

### 💻 Analysis 

The `mounted?` [method](https://github.com/inspec/inspec/blob/master/lib/resources/mount.rb#L30) of the `mount` resource doesn't check if `file` is a non NilClass object before calling its `mounted?` method. 

In the context of `inspec archive` the mount Class is not initialized meaning the `file` instance variable is not defined.  

This fixes by first checking if `file` is not `nil`:
```
    def mounted?
      file.nil? ? false : file.mounted?
    end
```

**Comment by clintoncwolfe** [comment]
Refs #2987

**Comment by jeremymv2** [comment]
Thanks @clintoncwolfe

**Comment by clintoncwolfe** [comment]
I've changed to the title to clarify that this is a more serious, widespread issue; not just affecting `mount`.

**Comment by clintoncwolfe** [comment]
Closing, believed fixed on #3126.  Please re-open if mistaken.
